pkg/apiserver/api_installer.go

```diff
@@ -110,62 +111,55 @@ func (a *APIInstaller) NewWebService() *restful.WebService {
 	return ws
 }
 
-func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storage, ws *restful.WebService, proxyHandler http.Handler) (*unversioned.APIResource, error) {
-	admit := a.group.Admit
-	context := a.group.Context
-
-	optionsExternalVersion := a.group.GroupVersion
-	if a.group.OptionsExternalVersion != nil {
-		optionsExternalVersion = *a.group.OptionsExternalVersion
-	}
-
-	var resource, subresource string
-	switch parts := strings.Split(path, "/"); len(parts) {
-	case 2:
-		resource, subresource = parts[0], parts[1]
-	case 1:
-		resource = parts[0]
-	default:
-		// TODO: support deeper paths
-		return nil, fmt.Errorf("api_installer allows only one or two segment paths (resource or resource/subresource)")
+// validFullyQualifiedKind validates and returns the group version kind registered for the given
+// storage object. If the storage object is a subresource, it just return's the object's group
+// version kind without validating it's group against the group we are installing the REST
+// handlers for here.
+func (a *APIInstaller) validFullyQualifiedKind(path string, storage rest.Storage, hasSubresource bool) (unversioned.GroupVersionKind, error) {
```

Comment:
Suggest: getBodyKind, getWireKind, getResourceKind

The validation isn't important to highlight in the function name. The thing to call attention to is that this produces the kind corresponding to the return type, not the kind corresponding to the path the resource is mounted under.